File: java/src/NthSeries.java
import java.text.DecimalFormat;
import java.util.stream.DoubleStream;

public class NthSeries {
    public static String seriesSum(int n) {
       double res = DoubleStream.iterate(1, i -> i + 3.0)
                .limit(n)
                .reduce(0, (prev, cur) -> prev + (1 / cur));

       DecimalFormat df = new DecimalFormat("0.00");

        return df.format(res);
    }

    public static void main(String[] args) {
        System.out.println(seriesSum(3));
    }
}

File: python/nth-series.py
def series_sum(n):
    def generate_sequence():
        i = 1
        for _ in range(0, n):
            yield i
            i = i + 3.0
    return "{0:.2f}".format(sum(1.0 / val for val in generate_sequence()))

if __name__ == '__main__':
    print(series_sum(3))
